### ISSUE TITLE
Add context to BaggageClaim API calls

### ISSUE THREAD

**Issue by ddadlani** [concourse/concourse#4166]
Similar to #3928, we should also cancel streaming in/out when the parent context is cancelled.

**Comment by kcmannem** [comment]
I'd like to add idle timeouts to our http client used in the baggageclaim client.

```
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: (&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
		DualStack: true,
	}).DialContext,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
```
The default transport allows you to set `MaxIdleConns` and `IdleConnTimeout` but we use our retryhttp transport object so we'll need to pass it in from there.

**Comment by kcmannem** [comment]
well I don't think idleConnTimeout is what I think it is. This is a great post explaining everything http://tleyden.github.io/blog/2016/11/21/tuning-the-go-http-client-library-for-load-testing/

**Comment by kcmannem** [comment]
We tried to check if there is a way to natively implement Write/Read Deadlines such that during a long running response recv. We can timeout when a period of the response body transmission connection remains in an idle state. We do this with the tsaClient like so https://github.com/concourse/concourse/blob/671e8f085b3e0c6b1af23462b3c59cf1c0b67dcf///tsa/timeout_conn.go#L8:6 but we decided to avoid doing this for now until we investigate further. We will only add the contexts for now.
